newIDE/app/src/EventsFunctionsExtensionsLoader/MetadataDeclarationHelpers.js

```diff
@@ -910,37 +1140,69 @@ export const declareObjectPropertiesInstructionAndExpressions = (
  */
 export const declareEventsFunctionParameters = (
+  eventsFunctionsContainer: gdEventsFunctionsContainer,
   eventsFunction: gdEventsFunction,
-  instructionOrExpression: gdInstructionMetadata | gdExpressionMetadata
+  instructionOrExpression:
+    | gdInstructionMetadata
+    | gdExpressionMetadata
+    | gdMultipleInstructionMetadata,
+  userDefinedFirstParameterIndex: number
 ) => {
-  mapVector(
-    eventsFunction.getParameters(),
-    (parameter: gdParameterMetadata) => {
-      if (!parameter.isCodeOnly()) {
-        instructionOrExpression.addParameter(
-          parameter.getType(),
-          parameter.getDescription(),
-          '', // See below for adding the extra information
-          parameter.isOptional()
-        );
-        instructionOrExpression.setParameterLongDescription(
-          parameter.getLongDescription()
-        );
-        instructionOrExpression.setDefaultValue(parameter.getDefaultValue());
-      } else {
-        instructionOrExpression.addCodeOnlyParameter(
-          parameter.getType(),
-          '' // See below for adding the extra information
-        );
-      }
-      // Manually add the "extra info" without relying on addParameter (or addCodeOnlyParameter)
-      // as these methods are prefixing the value passed with the extension namespace (this
-      // was done to ease extension declarations when dealing with object).
-      instructionOrExpression
-        .getParameter(instructionOrExpression.getParametersCount() - 1)
-        .setExtraInfo(parameter.getExtraInfo());
+  const addParameter = (parameter: gdParameterMetadata) => {
+    if (!parameter.isCodeOnly()) {
+      instructionOrExpression.addParameter(
+        parameter.getType(),
+        parameter.getDescription(),
+        parameter.getExtraInfo(),
+        parameter.isOptional()
+      );
+      instructionOrExpression.setParameterLongDescription(
+        parameter.getLongDescription()
+      );
+      instructionOrExpression.setDefaultValue(parameter.getDefaultValue());
+    } else {
+      instructionOrExpression.addCodeOnlyParameter(
+        parameter.getType(),
+        parameter.getExtraInfo()
+      );
     }
-  );
+  };
+
+  const functionType = eventsFunction.getFunctionType();
+  const getterFunction = eventsFunctionsContainer.hasEventsFunctionNamed(
+    eventsFunction.getGetterName()
+  )
+    ? eventsFunctionsContainer.getEventsFunction(eventsFunction.getGetterName())
+    : null;
+  // This is used instead of getParametersForEvents because the Value parameter
+  // is already add by useStandardOperatorParameters.
+  const parameters = (functionType === gd.EventsFunction.ActionWithOperator &&
+  getterFunction
+    ? getterFunction
+    : eventsFunction
+  ).getParameters();
+
+  mapVector(parameters, addParameter, 0, userDefinedFirstParameterIndex);
+
+  if (functionType === gd.EventsFunction.ExpressionAndCondition) {
+    ((instructionOrExpression: any): gdMultipleInstructionMetadata).useStandardParameters(
```

Comment:
Do you think we should have been naming this `useStandardOperatorParameters`?

Reply:
Probably, I can create a card for it.